src/views/components/SharePanel/index.jsx

```diff
@@ -43,8 +43,15 @@ const SubscriptionForm = ({ subscribe, status, message }) => {
   const [email, setEmail] = useState('');
   const [name, setName] = useState('');
   const [throttled, setThrottled] = useState(false);
+  const [unthrottleTimeout, setUnthrottleTimeout] = useState(null);
   const navigate = useNavigate();
   const navigateToHome = useCallback(() => navigate(HOME), [navigate]);
+  useEffect(() => {
+    if (status === "success") navigateToHome();
+    return () => {
+      if (unthrottleTimeout != null) clearTimeout(unthrottleTimeout);
+    };
```

Comment:
When a useEffect returns a function, it will get called when the component gets unmounted. Here we make sure when we unmount the component, we also clear any remaining timeout.